### PR TITLE
Dev-1391

### PR DESCRIPTION
**Pull request by fabian-cellosign**
Dev-1391

### CODE REVIEW

File: tests/unittests/test_bank_vaildators.py [hunk 1]
```diff
@@ -0,0 +1,261 @@
+import unittest
```
Comment (fabian-cellosign) [inline tests/unittests/test_bank_vaildators.py]: Please add allure for prints:
```
@allure.feature('Feature name ...')
@allure.title('Test of XYZ ...')
```

File: tests/unittests/test_bank_vaildators.py [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+
+from israel_bank_account_validator import BANK_VALIDATORS, SUPPORTED_BANKS, number_digits_to_list
+
+
+class TestBankValidators(unittest.TestCase):
+    def test_yahav_validator(self):
+        # Account number: 760050, Branch number: 482, should be valid
+        self.assertTrue(BANK_VALIDATORS[SUPPORTED_BANKS['YAHAV']](482, number_digits_to_list(760050, 6),
```
Comment (fabian-cellosign) [inline tests/unittests/test_bank_vaildators.py]: Please avoid repetition on nested properties:
`self.assertTrue(BANK_VALIDATORS[SUPPORTED_BANKS['YAHAV']](...`

```
validator = BANK_VALIDATORS[SUPPORTED_BANKS['YAHAV']]
self.assertTrue(validator(...
```

File: israel_bank_account_validator.py [hunk 3]
```diff
@@ -1,168 +1,309 @@
...
+MIZRAHI_TEFAHOT_BRANCH_THRESHOLD = 400
+MIZRAHI_TEFAHOT_BRANCH_THRESHOLD_MIN = 401
+MIZRAHI_TEFAHOT_BRANCH_THRESHOLD_MAX = 799
+
+LEUUMI_BRANCH_THRESHOLD = 800
+
+
+def convert_to_int(input_value):
+    if isinstance(input_value, str):
+        return int(input_value)
```
Comment (fabian-cellosign) [inline israel_bank_account_validator.py]: Please add try catch

File: israel_bank_account_validator.py [hunk 4]
```diff
@@ -1,168 +1,309 @@
...
+
+def scalar_product(arr1, arr2):
+    return sum(a * b for a, b in zip(arr1, arr2))
+
+
+def number_digits_to_list(num, length):
+    return [int(x) for x in str(num).zfill(length)]
+
+
+def is_non_negative_integer(num):
```
Comment (fabian-cellosign) [inline israel_bank_account_validator.py]: Please rename to `is_positive_integer`

File: tests/unittests/test_bank_account_validation.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
+import unittest
```
Comment (fabian-cellosign) [inline tests/unittests/test_bank_account_validation.py]: Please add Allure decorators

File: tests/unittests/test_bank_validators.py [hunk 6]
```diff
@@ -0,0 +1,280 @@
+import unittest
```
Comment (fabian-cellosign) [inline tests/unittests/test_bank_validators.py]: Please add Allure decorators

File: israel_bank_account_validator.py [hunk 7]
```diff
@@ -1,168 +1,309 @@
...
+        raise BankNumberValueError("Bank number is not a non-negative integer")
+
+    # Check if bank is supported
+    if bank_number not in SUPPORTED_BANKS.values():
+        logger.error("Unsupported bank number")
+        raise UnsupportedBankError("Unsupported bank number")
+
     if not is_non_negative_integer(branch_number):
-        return False
+        raise BankBranchNumberValueError("Branch number is not a non-negative integer")
```
Comment (fabian-cellosign) [inline israel_bank_account_validator.py]: raise BankBranchNumberValueError("Branch number is not valid negative integer")

File: israel_bank_account_validator.py [hunk 8]
```diff
@@ -1,168 +1,309 @@
...
     if not is_non_negative_integer(account_number):
-        return False
+        raise BankAccountNumberValueError("Account number is not a non-negative integer")
```
Comment (fabian-cellosign) [inline israel_bank_account_validator.py]: raise BankBranchNumberValueError("Account number is not valid negative integer")

File: israel_bank_account_validator.py (hunk 3 above)
Comment (napso20) [reply israel_bank_account_validator.py:38]: @fabian-cellosign We have exception handling on the caller side and it's in purpose: https://github.com/republixapps/israel-bank-account-validator/pull/1/files#diff-82c6de86245fefe2ebdd0cfaa669bc7e97e658324f9f264516ff71115e9ca636R82

**Pull request merged**